janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -7409,4 +7409,20 @@ public void testVerticesDropAfterWhereWithBatchQueryEnabled() {
         assertFalse(tx.traversal().V().has("a", timestamp).has("b", true).has("c", true).bothE().hasNext());
     }
 
+    @Test
+    public void testPropertiesAfterProjectWithByThenChoose() {
+        clopen(option(USE_MULTIQUERY),true,
+            option(BATCH_PROPERTY_PREFETCHING),false,
+            option(STORAGE_BATCH),false
+        );
+
+        long timestamp = System.currentTimeMillis();
+
+        tx.traversal().addV("test").property("a", timestamp).next();
+
+        newTx();
+
+        assertTrue(tx.traversal().V().has("a", timestamp).project("a").by(__.choose(__.has("a"), __.values("a"))).hasNext());
```

Comment:
I guess this check could be without `choose` step but this check with `choose` step looks good as well.
```
tx.traversal().V().has("a", timestamp).project("a").by(__.has("a").values("a")).hasNext()
```